ob1k-consul/src/main/java/com/outbrain/ob1k/consul/LoadBalancer.java

```diff
@@ -0,0 +1,10 @@
+package com.outbrain.ob1k.consul;
+
+import java.util.List;
+
+public interface LoadBalancer {
```

Comment:
I feel like the abstraction is leaky here - as you have both #onTargetsChanged method that receives updates for changed targets (which is only relevant to dynamic providers, such as Consul), and also #provideTargets receives the current state of the available nodes.

Try to think how to make the load balancer implementation to be stateless - and it will help you to simplify the abstraction. One idea, that I'm a bit stealing here from gRPC, is to wrap the server into a more complex value object that contains attributes - which means you can mark each node itself and maintain state of the node rather inside the LB implementation (for example: https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/util/RoundRobinLoadBalancerFactory.java)